### ISSUE TITLE
armresources fails to compile

### ISSUE THREAD

**Issue by ItalyPaleAle** [Azure/azure-sdk-for-go#17476]
### Bug Report

Trying to run the code from this sample: https://docs.microsoft.com/en-us/azure/developer/go/deploy-azure-resource-manager-template

Copy/pasted as-is, and the code doesn't compile: 

```sh
$ go run .
# github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources
../../../../go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources@v0.3.1/zz_generated_pollers.go:19:13: undefined: azcore.Poller
../../../../go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources@v0.3.1/zz_generated_pollers.go:62:13: undefined: azcore.Poller
../../../../go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources@v0.3.1/zz_generated_pollers.go:105:13: undefined: azcore.Poller
../../../../go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources@v0.3.1/zz_generated_pollers.go:148:13: undefined: azcore.Poller
../../../../go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources@v0.3.1/zz_generated_pollers.go:191:13: undefined: azcore.Poller
../../../../go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources@v0.3.1/zz_generated_pollers.go:234:13: undefined: azcore.Poller
../../../../go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources@v0.3.1/zz_generated_pollers.go:277:13: undefined: azcore.Poller
../../../../go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources@v0.3.1/zz_generated_pollers.go:320:13: undefined: azcore.Poller
../../../../go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources@v0.3.1/zz_generated_pollers.go:363:13: undefined: azcore.Poller
../../../../go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources@v0.3.1/zz_generated_pollers.go:406:13: undefined: azcore.Poller
../../../../go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources@v0.3.1/zz_generated_pollers.go:406:13: too many errors
```

Go version: `go version go1.18 darwin/amd64`

Relevant deps and versions:

	github.com/Azure/azure-sdk-for-go/sdk/azidentity v0.14.0
	github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources v0.3.1
	github.com/Azure/azure-sdk-for-go/sdk/azcore v0.23.0 // indirect
	github.com/Azure/azure-sdk-for-go/sdk/internal v0.9.2 // indirect
	github.com/AzureAD/microsoft-authentication-library-for-go v0.4.0 // indirect

**Comment by chlowell** [comment]
Thanks for opening this issue. Sorry the article has a major pitfall now. It's fallen into the gap between two waves of our releases. We released azcore v0.23.0 earlier this week, with some breaking changes vs. 0.22.0, and a new azidentity that requires it. However, we haven't yet released an updated armresources, so the latest version of that module isn't compatible with the latest azcore. If you were following the sample, you ran `go get azidentity` and ended up with the latest version of it and azcore, leading to this problem. We'll release a compatible armresources soon. In the meantime, you can work around this by downgrading to azcore v0.22.0 and azidentity v0.13.2.

**Comment by ItalyPaleAle** [comment]
Thanks for responding!

I did try downgrading to the prior versions and thankfully it does work now.

**Comment by ItalyPaleAle** [comment]
@chlowell Do you know when the updated "armresources" SDK will be released?

I am now faced with an issue due to the fact that another SDK I need (the for-now named "armapp" that @tadelesh is working on) requires azcore 0.23, and I can't use that or it will break "armresources".

**Comment by chlowell** [comment]
It's available now ([v0.4.0](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources))

**Comment by ItalyPaleAle** [comment]
Great, thank you!
